src/codegen/codegen_neuron_cpp_visitor.cpp

```diff
@@ -89,36 +92,281 @@ void CodegenNeuronCppVisitor::print_atomic_reduction_pragma() {
 /****************************************************************************************/
 
 
+void CodegenNeuronCppVisitor::print_point_process_function_definitions() {
+    if (info.point_process) {
+        printer->add_line("/* Point Process specific functions */");
+        printer->add_multi_line(R"CODE(
+            static void* _hoc_create_pnt(Object* _ho) {
+                return create_point_process(_pointtype, _ho);
+            }
+        )CODE");
+        printer->push_block("static void _hoc_destroy_pnt(void* _vptr)");
+        if (info.is_watch_used() || info.for_netcon_used) {
+            printer->add_line("Prop* _prop = ((Point_process*)_vptr)->prop;");
+        }
+        if (info.is_watch_used()) {
+            printer->push_block("if (_prop)");
+            printer->fmt_line("_nrn_free_watch(_nrn_mechanism_access_dparam(_prop), {}, {});",
+                              info.watch_count,
+                              info.is_watch_used());
+            printer->pop_block();
+        }
+        if (info.for_netcon_used) {
+            printer->push_block("if (_prop)");
+            printer->fmt_line(
+                "_nrn_free_fornetcon(&(_nrn_mechanism_access_dparam(_prop)[_fnc_index].literal_"
+                "value<void*>()));");
+            printer->pop_block();
+        }
+        printer->add_line("destroy_point_process(_vptr);");
+        printer->pop_block();
+        printer->add_multi_line(R"CODE(
+            static double _hoc_loc_pnt(void* _vptr) {
+                return loc_point_process(_pointtype, _vptr);
+            }
+        )CODE");
+        printer->add_multi_line(R"CODE(
+            static double _hoc_has_loc(void* _vptr) {
+                return has_loc_point(_vptr);
+            }
+        )CODE");
+        printer->add_multi_line(R"CODE(
+            static double _hoc_get_loc_pnt(void* _vptr) {
+                return (get_loc_point_process(_vptr));
+            }
+        )CODE");
```

Comment:
I combined some of them related to this PR. I think there are some more opportunities to combine `add_line`s and `add_multi_line`s across the `CodegenNeuronCppVisitor` but maybe this can come in another PR